app/models/event_preview.rb

```diff
@@ -0,0 +1,22 @@
+class EventPreview < ApplicationRecord
+  default_scope { order('created_at DESC') }
+
+  has_one :image, as: :imageable, dependent: :destroy
+  belongs_to :event, foreign_key: :year, primary_key: :year, touch: true
+      
+      accepts_nested_attributes_for :image, reject_if: :all_blank, allow_destroy: true
```

Comment:
In this repo, we usually use two spaces for indentation because '\t' or tab character can be translated into different amount of white spaces